osu.Game/Rulesets/Objects/SliderPathExtensions.cs

```diff
@@ -25,6 +29,71 @@ public static void SnapTo<THitObject>(this THitObject hitObject, IDistanceSnapPr
         /// <param name="sliderPath">The <see cref="SliderPath"/>.</param>
         /// <param name="positionalOffset">The positional offset of the resulting path. It should be added to the start position of this path.</param>
         public static void Reverse(this SliderPath sliderPath, out Vector2 positionalOffset)
+        {
+            var controlPoints = sliderPath.ControlPoints;
+            var originalControlPointTypes = controlPoints.Select(p => p.Type).ToArray();
+
+            controlPoints[0].Type ??= PathType.Linear;
+
+            // Inherited points after a linear point should be treated as linear points.
+            controlPoints.Where(p => sliderPath.PointsInSegment(p)[0].Type == PathType.Linear).ForEach(p => p.Type = PathType.Linear);
```

Comment:
This fix looks rather heavy-handed and it seems like it will create many more non-inherited control points than strictly necessary. Can you explain why this is necessary? Which cases are broken by not doing this? Is this fix covered by any of the added test cases?

Reply:
Okay, I see. I'd say two things as a follow-up to that:

- The "restoration" of control point types in the middle of the method confused me but I figured I'd leave that be at that point. It confused me mostly because it affected all of the points, but I assume it is done explicitly for this fix. If that is the case, I'd prefer that points which were temporarily modified to linear were collected to a list and then reverted from that list. (Or at least add an inline comment before restoring the control points that explains _why_ the control points are being reverted.)
- If possible I'd like to see test coverage that checks control point types too, to ensure the inherited linear points stay inherited at the end of it all. Up to you whether you wanna try, I may do this myself in the end.